db/index.js

```diff
@@ -837,6 +837,11 @@ function queryTable(store, table, data, opts, isLocal, fetchFromItemDb, startKey
 
   var queryFilter = data.QueryFilter || data.ScanFilter
 
+  var preFilteredItems = []
+  vals.join(function(items) {
```

Comment:
If we're doing a `join` here, then when don't need to do another join down below – all subsequent operations can operate on the final array.